### PR TITLE
[risk=low][RW-10991] Split out some components from runtime-config-panel

### PR DESCRIPTION
**Pull request by jmthibault79**
Split these components into their own files:
- CommonButton (new)
- CreatePanel
- PresetSelector

No expected changes in functionality.

Tested by running locally and observing:
- buttons work as expected
- presets work as expected
- creation works as expected

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/runtime-configuration-panel.tsx [hunk 1]
```diff
@@ -491,83 +385,59 @@ const PanelMain = fp.flow(
...
-          onClick={() => {
-            requestAnalysisConfig(analysisConfig);
-            onClose();
-          }}
-        >
-          {updateMessaging.applyAction}
-        </Button>
-      );
-    };
+    const renderCreateOrUpdateButton = (
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel.tsx:388]: alt: remove these render functions entirely and specify the components inline

File: ui/src/app/components/runtime-configuration-panel/preset-selector.tsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    <Dropdown
+      id='runtime-presets-menu'
+      appendTo='self'
+      disabled={disabled}
+      style={{
+        marginTop: '21px',
+        color: colors.primary,
+      }}
+      placeholder='Recommended environments'
+      options={Object.values(runtimePresets)
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel/preset-selector.tsx:33]: this is changed from the original: I converted the lodash/fp code to the equivalent plain JS

File: ui/src/app/components/runtime-configuration-panel/preset-selector.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  gcePersistentDisk: Disk | null | undefined;
+}
+// Select a recommended preset configuration.
+export const PresetSelector = ({
+  allowDataproc,
+  setAnalysisConfig,
+  disabled,
+  gcePersistentDisk,
+}: PresetSelectorProps) => {
+  const options = Object.values(runtimePresets)
```
Comment (jmthibault79) [inline ui/src/app/components/runtime-configuration-panel/preset-selector.tsx:23]: this was originally set using lodash/fp, but that's not necessary

File: ui/src/app/components/runtime-configuration-panel/common-button.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import { Button } from 'app/components/buttons';
+
+interface Props {
+  label: string;
+  buttonText?: string;
+  onClick: () => void;
+  disabled: boolean;
+}
+export const CommonButton = ({
```
Comment (Peter-Lavigne) [inline ui/src/app/components/runtime-configuration-panel/common-button.tsx:11]: I find having both a "CommonButton" and "Button" confusing. My preference would be to either:
- Simplify the `Button` Component such that this isn't needed or
- Leave this as a private component in the original file so we don't end up with two publicly-scoped Buttons

**Review by Peter-Lavigne — APPROVED** [review]
LGTM. Consider both of my comments optional, as this PR is definitely an improvement from before. I like the approach of extracting, testing, and understanding isolated components.

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel.tsx:388]: Do you mean to say that these would work better as proper components?

File: ui/src/app/components/runtime-configuration-panel/common-button.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/components/runtime-configuration-panel/common-button.tsx]: Makes sense - I put CommonButton back into R-C-P

File: ui/src/app/components/runtime-configuration-panel.tsx (hunk 1 above)
Comment (Peter-Lavigne) [reply ui/src/app/components/runtime-configuration-panel.tsx:404]: Yes, that's a better way to word it

**Pull request merged**